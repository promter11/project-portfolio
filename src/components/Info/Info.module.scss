@import "../../styles/_variables";

.info {
  .block {
    display: flex;
    align-items: center;

    &:not(:last-child) {
      margin-bottom: 3rem;
    }

    .image {
      display: block;
      width: 3rem;
      height: 3rem;
      margin-right: 1.5rem;
    }

    .text {
      font-size: 1.4rem;
      font-weight: 100;
      line-height: 1.8rem;
      color: $white-color;
    }
  }
}
