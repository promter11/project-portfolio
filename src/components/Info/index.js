import React from "react";

import styles from "./Info.module.scss";

const Info = ({ items }) => {
  return (
    <div className={styles.info}>
      {items.map(({ image, value, path }, index) => {
        return (
          <div key={index} className={styles.block}>
            <img className={styles.image} src={image} alt="Info" />
            {path ? (
              <a className={styles.text} href={path}>
                {value}
              </a>
            ) : (
              <span className={styles.text}>{value}</span>
            )}
          </div>
        );
      })}
    </div>
  );
};

export default Info;
