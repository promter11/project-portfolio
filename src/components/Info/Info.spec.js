import React from "react";
import { shallow } from "enzyme";

import Info from "./index";

const items = [
  {
    image: "../images/info/1",
    value: "Test",
    path: "/path",
  },
  {
    image: "../images/info/2",
    value: "Test 2",
    path: "/link",
  },
  {
    image: "../images/info/3",
    value: "Test 3",
  },
];

describe("Info component", () => {
  it("renders correctly with items", () => {
    const wrapper = shallow(<Info items={items} />);

    expect(wrapper).toMatchSnapshot();
  });
});
