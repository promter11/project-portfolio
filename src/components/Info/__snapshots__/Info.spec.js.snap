// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component renders correctly with items 1`] = `
<div
  className="info"
>
  <div
    className="block"
    key="0"
  >
    <img
      alt="Info"
      className="image"
      src="../images/info/1"
    />
    <a
      className="text"
      href="/path"
    >
      Test
    </a>
  </div>
  <div
    className="block"
    key="1"
  >
    <img
      alt="Info"
      className="image"
      src="../images/info/2"
    />
    <a
      className="text"
      href="/link"
    >
      Test 2
    </a>
  </div>
  <div
    className="block"
    key="2"
  >
    <img
      alt="Info"
      className="image"
      src="../images/info/3"
    />
    <span
      className="text"
    >
      Test 3
    </span>
  </div>
</div>
`;
